src/harness/unittests/telemetry.ts

```diff
@@ -255,6 +255,17 @@ namespace ts.projectSystem {
             return events;
         }
 
+        getEventsWithName<T extends server.ProjectServiceEvent>(eventName: T["eventName"]): ReadonlyArray<T> {
+            let events: T[];
+            removeWhere(this.events, event => {

```

Comment:
Use `filterMutate` instead of `removeWhere` (#19082). But does this function need to mutate `this.events`?

Reply:
Yes that function was written such that the already queried events would be removed.